### ISSUE TITLE
Attempting to save custom clases causes the json file to break

### ISSUE THREAD

**Issue by scj643** [controversial/livejson#4]
``` python
import livejson
class Test (object):
    def __init__(self):
        value = 1


f = livejson.File("test.json")
f['test']= Test()
```

Will cause the file test.json to become unreadable by livejson.

**Comment by controversial** [comment]
Thanks for the report. I'll work on a fix.

Right now, I'm thinking I'll
1. Cache the existing value before write
2. Try to update the database
3. If the write fails, catch the error, then rollback before re-raising the error

I'll be able to work on this in about an hour

**Comment by controversial** [comment]
Closed via 08597ceffeaf4613eedcece1a63e8c601eb49598.
